formatters/formatter.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+class Formatter {
+  static get FORMATS() {
+    if (!this._formatters) {
```

Comment:
The Formatter class is. We basically "annotate" the extendedInfo with the correct type so that the pretty and HTML outputs can defer to it for rendering:

```javascript
return ARIAValidAttr.generateAuditResult({
      value: typeof rule === 'undefined',
      debugString: this.createDebugString(rule),
      extendedInfo: {
        formatter: Formatter.SUPPORTED_FORMATS.ACCESSIBILITY,
        value: rule
      }
    });
```

In the PR there's only one formatter (L35 of formatter.js), which is a11y data. That said, we should expect more of these because of the various diagnoses coming (CRP, dependency chain)